src/main/java/org/spongepowered/api/map/MapCanvas.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map;
+
+import com.flowpowered.math.vector.Vector2i;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.data.persistence.InvalidDataException;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.map.color.MapColor;
+import org.spongepowered.api.map.color.MapColorTypes;
+
+import java.awt.Image;
+import java.util.Optional;
+
+/**
+ * A {@code MapCanvas} represents the image that is drawn on a
+ * {@link ItemTypes#FILLED_MAP}.
+ *
+ * <p>A canvas in vanilla Minecraft is 128x128 pixels.</p>
+ */
+public interface MapCanvas extends DataSerializable {
+
+    /**
+     * Creates a builder that creates a new canvas.
+     *
+     * @return A {@link Builder} to generate a new canvas.
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Gets the {@link MapColor} at the requested location.
+     *
+     * @param x The x location
+     * @param y The y location
+     * @return MapColor color at this location
+     * @throws IllegalArgumentException if either of the requested
+     *      co-ordinates are out of bounds.
```

Comment:
What are the bounds? Where is (0, 0)?

[Take a look at how I've described it here in `MapDecoration`](https://github.com/SpongePowered/SpongeAPI/pull/2112/files#diff-93c4ab7cf6dd5ffeb933fbd0791f389bR36) - this is the sort of information a user will need.